influxdb3_write/src/parquet_cache/mod.rs

```diff
@@ -175,56 +331,17 @@ const STORE_NAME: &str = "mem_cached_object_store";
 pub struct MemCachedObjectStore {
     /// An inner object store for which items will be cached
     inner: Arc<dyn ObjectStore>,
-    /// A weighted LRU cache for storing the objects associated with a given path in memory
-    // NOTE(trevor): this uses a mutex as the CLruCache type needs &mut self for its get method, so
-    // we always need an exclusive lock on the cache. If this creates a performance bottleneck then
-    // we will need to look for alternatives.
-    //
-    // A Tokio mutex is used to prevent blocking the thread while waiting for a lock, and so that
-    // the lock can be held accross await points.
-    cache: Arc<Mutex<CLruCache<Path, CacheEntry, RandomState, CacheEntryScale>>>,
+    cache: Arc<Cache>,
 }
 
 impl MemCachedObjectStore {
     /// Create a new [`MemCachedObjectStore`] with the given memory capacity
```

Comment:
Maybe a comment here on the fact that it is fixing the max freed amount at 10% of the capacity.